src/components/CalendarPicker/index.js

```diff
@@ -0,0 +1,163 @@
+import _ from 'underscore';
+import React from 'react';
+import {View, TouchableOpacity} from 'react-native';
+import moment from 'moment';
+import Text from '../Text';
+import ArrowIcon from './ArrowIcon';
+import styles from '../../styles/styles';
+import {propTypes as calendarPickerPropType, defaultProps as defaultCalendarPickerPropType} from './calendarPickerPropTypes';
+import generateMonthMatrix from './generateMonthMatrix';
+import withLocalize from '../withLocalize';
+import Navigation from '../../libs/Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+import CONST from '../../CONST';
+
+class CalendarPicker extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.monthNames = moment.localeData(props.preferredLocale).months();
+        this.daysOfWeek = moment.localeData(props.preferredLocale).weekdays();
+
+        let currentDateView = props.value;
+        if (props.maxDate < currentDateView) {
+            currentDateView = props.maxDate;
+        } else if (props.minDate > currentDateView) {
+            currentDateView = props.minDate;
+        }
+
+        if (props.defaultYear) {
+            currentDateView = moment(currentDateView).set('year', props.defaultYear).toDate();
+        }
```

Comment:
```suggestion
        if (props.defaultYear) {
            currentDateView = moment(currentDateView).set('year', props.defaultYear).toDate();
        }

        if (props.maxDate < currentDateView) {
            currentDateView = props.maxDate;
        } else if (props.minDate > currentDateView) {
            currentDateView = props.minDate;
        }
```

Those should be swapped. Checking for min and max should be the last step.